### PR TITLE
Allow admins to redact a PDF - Part 1

### PR DESCRIPTION
**Pull request by FerPerales**
Asana ticket: https://app.asana.com/0/1203289004376659/1203289004376687/f

image

image

First part of the feature:

* Integrate Foxit SDK
* Add Redact document in source file when a document is a PDF file

### CODE REVIEW

File: app/models/source_file.rb [hunk 1]
```diff
@@ -34,4 +34,8 @@ def claimed?
   def standards_import
     active_storage_attachment.record
   end
+
+  def pdf?
+    active_storage_attachment&.filename&.extension == "pdf"
```
Comment (littleforest) [inline app/models/source_file.rb:39]: What about calling content type on the attachment? `active_storage_attachment.blob.content_type == "application/pdf"`

File: config/environments/development.rb [hunk 2]
```diff
@@ -64,6 +64,8 @@
   # Enable debug mode in development
   config.assets.debug = true
 
+  config.assets.compile = true
```
Comment (littleforest) [inline config/environments/development.rb:67]: What are the implications of this for making changes to assets in development? Do we have to restart the server every time? Should we make this a conditional for if we have "work on pdf redaction tickets flag turned on"?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 Rails.application.routes.draw do
+  namespace :admin do
+    get "redact_files/new"
+  end
```
Comment (littleforest) [inline config/routes.rb:4]: Do we need both these lines and line 41 below since it is not an AR-backed resource?

**Review by littleforest — COMMENTED** [review]
This is great! Just a couple of small questions.

File: config/routes.rb (hunk 3 above)
Comment (FerPerales) [reply config/routes.rb:4]: @littleforest good catch! We only need this in line 41

File: config/environments/development.rb (hunk 2 above)
Comment (FerPerales) [reply config/environments/development.rb:67]: @littleforest We don't need this anymore. It was not the issue for missing assets

**Review by littleforest — APPROVED** [review]
🚢

**Pull request merged**